src/api/events/registerForEvents.js

```diff
@@ -0,0 +1,32 @@
+/* @flow */
+import type { Auth, ApiResponseSuccess, Narrow } from '../apiTypes';
+import type { InitialData } from '../../types';
+import { apiPost } from '../apiFetch';
+
+/** See https://zulipchat.com/api/register-queue */
+
+type ApiResponseRegister = ApiResponseSuccess & {
+  queue_id: string,
+  last_event_id: number,
+  data: InitialData,
+};
+
+export default (
+  auth: Auth,
+  eventTypes: string[],
+  fetchEventTypes: string[],
+  applyMarkdown: boolean = true,
+  includeSubscribers: boolean = false,
```

Comment:
This and `allPublicStreams` don't match the server defaults (according to the doc.) That's fine for now, but let's clearly mark that in jsdoc -- otherwise it's the kind of thing that could give someone a nasty surprise (or frustrating debugging session) when they look at the API doc, then use this because they believe it's a thin wrapper around the server API, and it doesn't do as they expect.

Reply:
More explicitly:
* For the commit that adds these options, keeping the values we're currently passing (which are different from the default) is absolutely the right thing; it just needs to be called out in jsdoc.
* It'd be a good followup (in a separate commit) to make this function take the same defaults as the server API, and adjust the caller(s) to pass the desired non-default values. Doesn't have to be in the same branch, though.